hyperbahn/advertise.go

```diff
@@ -79,16 +79,17 @@ func (c *Client) advertiseLoop() {
 
 		if err := c.sendAdvertise(); err != nil {
 			consecutiveFailures++
-			if consecutiveFailures >= maxAdvertiseFailures {
+			if consecutiveFailures >= maxAdvertiseFailures && c.opts.FailStrategy == FailStrategyFatal {
 				c.opts.Handler.OnError(ErrAdvertiseFailed{Cause: err, WillRetry: false})
-				if c.opts.FailStrategy == FailStrategyFatal {
-					c.tchan.Logger().Fatalf("Hyperbahn client registration failed: %v", err)
-				}
-				return
+				c.tchan.Logger().Fatalf("Hyperbahn client registration failed: %v", err)
 			}
-			c.tchan.Logger().Warnf("Hyperbahn client registration failed (will retry): %v", err)
+			c.tchan.Logger().Infof("Hyperbahn client registration failed (will retry): %v", err)
```

Comment:
With the current Sentry defaults, I'd say no...but there's no reason to pull that discussion into this diff. I'll go back to the previous behavior and address error/warning levels separately.